build-support/bin/readme_api.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
This file is a bit long, but very straightforward. The actual important part (that it conforms to the live readme.com API) is hard to test, but the doc generator still works, so at least that pudding is proven. And we had no tests for that before either.

Reply:
> is hard to test

At what point do you think it would be helpful to start adding automated tests? I think we'd have to do a lot of mocking, so probably depends on how stable we trust the Readme API to be?

I agree our logic keeps getting more and more complex (for good reason!)

Reply:
I just don't see how to usefully test this. If we manually mock the Readme API then it's a lot of effort, and we're still not really testing the right thing.

If they published their OpenAPI spec then there are tools to generate mocks from that. But AFAICT they do not publish it (although presumably they use it to generate their API docs, since that is a feature they offer to their users).